js-medical-institution.js

```diff
@@ -0,0 +1,158 @@
+class Pacijent {
+  constructor(ime, prezime, jmbg, broj_zdravstvenog_kartona) {
+    this.ime = ime;
+    this.prezime = prezime;
+    this.jmbg = jmbg;
+    this.broj_zdravstvenog_kartona = broj_zdravstvenog_kartona;
+    this.pregledi = [];
+    this.izabraniDoktor = null;
+    loguj("Kreiran pacijent", this.ime);
+  }
+
+  izborDoktora(doktor) {
+    if (!this.izabraniDoktor) {
+      this.izabraniDoktor = doktor;
+      doktor.pacijenti.push(this);
+      loguj("Izabrao doktora", this.ime);
+    }
+  }
+
+  obavljanjePregleda(tip) {
+    let preostaliPregledi = this.pregledi.filter(pregled => pregled.tip != tip);
+    let zakazanPregled = this.pregledi.find((pregled) => pregled.tip == tip);
+    this.pregledi = preostaliPregledi;
+    //console.log(zakazanPregled);
+    zakazanPregled.pregled();
+      loguj("Obavljen pregled", this.ime);
+   // console.log(zakazanPregled);
+  }
+}
+
+class Doktor {
+  constructor(ime, prezime, specijalnost) {
+    this.ime = ime;
+    this.prezime = prezime;
+    this.specijalnost = specijalnost;
+    this.pacijenti = [];
+     loguj("Kreiran doktor", this.ime);
+  }
+
+  zakazivanjePregleda(pacijent, tip, datum, vreme) {
+    loguj("Zakazao pregled doktor", this.ime);
+    if (tip == "krvni pritisak") {
+      let pregled = new KrvniPritisak(tip, datum, vreme);
+      pacijent.pregledi.push(pregled);
+    }
+
+    if (tip == "nivo secera u krvi") {
+      let pregled = new NivoSecera(tip, datum, vreme);
+      pacijent.pregledi.push(pregled);
+    }
+
+    if (tip == "nivo holesterola u krvi") {
+      let pregled = new NivoHolesterola(tip, datum, vreme);
+      pacijent.pregledi.push(pregled);
+    }
+  }
+}
+
+class TipPregleda {
+  constructor(tip, datum, vreme) {
+    this.tip = tip;
+    this.datum = datum;
+    this.vreme = vreme;
+  }
+
+  getRandomInt(min, max) {
+    min = Math.ceil(min);
+    max = Math.floor(max);
+    return Math.floor(Math.random() * (max - min) + min);
+  }
+
+}
+
+class KrvniPritisak extends TipPregleda {
+  constructor(tip, datum, vreme) {
+    super(tip, datum, vreme);
+    this.gornjaVrednost = null;
+    this.donjaVrednost = null;
+    this.puls = null;
+  }
+
+  pregled() {
+    this.gornjaVrednost = super.getRandomInt(105, 170);
```

Comment:
Ne koristi super jer je ova klasa nasledila TipPregleda i samim tim na sebi (i ako nema na sebi ima na prototipu) ima tu metodu. Dakle `this.getRandomInt(105, 170)` 
Izmeni to ako jos negde ima